packages/app/src/cli/services/dev/fetch.ts

```diff
@@ -207,19 +193,15 @@ interface FetchStoreByDomainOutput {
  * Returns the organization and the store based on passed domain
  * If a store with that domain doesn't exist the method returns undefined
  * @param orgId - Organization ID
- * @param token - Token to access partners API
  * @param shopDomain - shop domain fqdn
+ * @param developerPlatformClient - The client to access the platform API
  */
 export async function fetchStoreByDomain(
   orgId: string,
-  token: string,
   shopDomain: string,
+  developerPlatformClient: DeveloperPlatformClient,
```

Comment:
Good point, I was aware of this issue and I had plans to fix it.

We have a mix of approaches in PartnersClient. For example:
- [orgFromId](https://github.com/Shopify/cli/blob/a4121b0da3e0b7abc051bf3fab38bf6a7a6d52d1/packages/app/src/cli/utilities/developer-platform-client/partners-client.ts#L132) calls a function from dev/fetch. There, we basically make the API request and check the response to raise an error if it fails.
- [storeByDomain](https://github.com/Shopify/cli/pull/3510/files#diff-5c8e4aee8775860dd20e55d6b85e4b71d3fe6167ea3c816b64449782682fa270R232) (added in this PR) just makes the API request. And then it's called from this function in dev/fetch, that handles the responses.

I think it's better to keep PartnersClient as simple as possible, just making the API requests. And then we can keep the error handling in the original functions. Otherwise, we would be coupling the API request to a logic related to dev. Right now, I think the errors are quite generic, but we could say something like `No org found. Please run dev again with this flag`. And that error would be shown if we make the same request from other command.

What do you think?

Reply:
I think your logic is sound 👍 

Better to return `undefined` and let the caller choose whether or not to handle it as an error.

Reply:
I'll tackle these changes in the next PR.